fedimint-cli/src/client.rs

```diff
@@ -489,6 +497,51 @@ pub async fn handle_command(
     }
 }
 
+async fn get_invoice(info: &str, amount: Option<Amount>) -> anyhow::Result<Bolt11Invoice> {
+    let info = info.trim();
+    match lightning_invoice::Bolt11Invoice::from_str(info) {
+        Ok(invoice) => {
+            debug!("Parsed parameter as bolt11 invoice: {invoice}");
+            match (invoice.amount_milli_satoshis(), amount) {
+                (Some(_), Some(_)) => {
+                    bail!("Amount specified in both invoice and command line")
+                }
```

Comment:
if these are equal it should be fine

Reply:
If the invoice doesn't have an amount the recipient won't enforce it, so we'd trust the gateway to actually send the whole amount. One day the client might do the routing, then this wouldn't be a problem anymore.

Reply:
Ah, sorry, totally missed that, good point!